tests/test_analysis.py

```diff
@@ -623,46 +647,214 @@ def helper_test_update_harmonic_vibe_props(analysis):
+            "Tutorial Project",
+            [
+                {
+                    "shock_result_count": 2,
+                    "critical_shock_strain": 10,
+                    "critical_shock_strain_units": "strain",
+                    "part_validation_enabled": True,
+                    "require_material_assignment_enabled": False,
+                    "force_model_rebuild": "AUTO",
+                    "natural_freq_min": 10,
+                    "natural_freq_min_units": "Hz",
+                    "natural_freq_max": 100,
+                    "natural_freq_max_units": "KHz",
+                    "analysis_temp": 20,
+                    "analysis_temp_units": "F",
+                },
+            ],
+        )
+        assert False
+    except SherlockUpdateMechanicalShockPropsError as e:
+        assert (
+            str(e) == "Update mechanical shock properties error: "
+            "CCA name is invalid for mechanical shock properties 0."
+        )
+
+    try:
+        analysis.update_mechanical_shock_props(
+            "Tutorial Project",
+            [
+                {
+                    "cca_name": "",
+                    "shock_result_count": 2,
+                    "critical_shock_strain": 10,
+                    "critical_shock_strain_units": "strain",
+                    "part_validation_enabled": True,
+                    "require_material_assignment_enabled": False,
+                    "force_model_rebuild": "AUTO",
+                    "natural_freq_min": 10,
+                    "natural_freq_min_units": "Hz",
+                    "natural_freq_max": 100,
+                    "natural_freq_max_units": "KHz",
+                    "analysis_temp": 20,
+                    "analysis_temp_units": "F",
+                },
+            ],
+        )
+        assert False
+    except SherlockUpdateMechanicalShockPropsError as e:
+        assert (
+            str(e) == "Update mechanical shock properties error: "
+            "CCA name is invalid for mechanical shock properties 0."
+        )
+
+    try:
+        result = analysis.update_mechanical_shock_props(
+            "Tutorial Project",
+            [
+                {
+                    "cca_name": "Main Board",
+                    "model_source": SherlockAnalysisService_pb2.ModelSource.GENERATED,
+                    "shock_result_count": 2,
+                    "critical_shock_strain": 10,
+                    "critical_shock_strain_units": "strain",
+                    "part_validation_enabled": True,
+                    "require_material_assignment_enabled": False,
+                    "force_model_rebuild": "AUTO",
+                    "natural_freq_min": 10,
+                    "natural_freq_min_units": "Hz",
+                    "natural_freq_max": 100,
+                    "natural_freq_max_units": "KHz",
+                    "analysis_temp": 20,
+                    "analysis_temp_units": "F",
+                },
+            ],
+        )
```

Comment:
This test should only run when connected.
Validate the result value indicates that the update was successful here instead of after checking if connected.
Move this to below the test that validates the exception type since that is a prerequisite for catching this exception.